aoc-client/src/lib.rs

```diff
@@ -421,6 +421,76 @@ impl AocClient {
         Ok(())
     }
 
+    fn get_personal_stats_html(&self) -> AocResult<String> {
+        debug!("🦌 Fetching {} personal stats", self.year);
+
+        let url =
+            format!("https://adventofcode.com/{}/leaderboard/self", self.year);
+        let response = http_client(&self.session_cookie, "text/html")?
+            .get(url)
+            .send()?;
+
+        if response.status() == StatusCode::NOT_FOUND {
+            // A 402 reponse means the calendar for
+            // the requested year is not yet available
+            return Err(AocError::InvalidEventYear(self.year));
+        } else if response.status() == StatusCode::FOUND {
+            // A 302 reponse is a redirect and it likely
+            // means we're not logged in
+            warn!(
+                "🍪 It looks like you are not logged in, try logging in again"
+            );
+            return Err(AocError::AocResponseError);
+        }
```

Comment:
Maybe this deserves a new error, say AocError::NotLoggedIn. When showing the calendar not being logged in isn't a problem as you can still show the calendar with no stars. But here it's different...